build_tools/third_party/swiftshader/build_vk_swiftshader.sh

```diff
@@ -41,12 +44,27 @@ if [[ $(basename "$PWD") != "iree" ]]; then
     exit 1
 fi
 
+#  Clone swiftshader and checkout the appropriate commit.
+if [[ ! -d "/tmp/swiftshader_install" ]]; then
+  git clone https://github.com/google/swiftshader "${SWIFTSHADER_DIR?}"
+fi
+IREE_DIR=$(pwd)
+cd "${SWIFTSHADER_DIR?}"
+git pull origin master --ff-only
+git checkout "${SWIFTSHADER_COMMIT?}"
+cd "${IREE_DIR?}"
+
 # Generate build system in build-swiftshader/ for third_party/swiftshader/.
 #
 # Options:
 #   - 64 bit platform and host compiler
 #   - Build Vulkan only, don't build GL
 #   - Don't build samples or tests
+
+if [[ -d "build-swiftshader" ]]; then
+  rm -rf build-swiftshader/
+fi
+
 cmake -B build-swiftshader/ \
```

Comment:
Should we put this somewhere different? Having it in-tree in IREE seems a bit weird and ties it to a specific git repo, which isn't necessary. We just need to set `VK_ICD_FILENAMES` to point at wherever it is, right?

Reply:
any suggestions? This is just where it was.

Reply:
Seems like asking the user where they want it would be reasonable

Reply:
Sounds good. I'll default it to `"${HOME?}/.swiftshader"`.

Reply:
I updated the documentation as well.

Reply:
This conversation seems to have moved to a different thread.